### DISCUSSION TITLE
How are files distributed across multiple volume servers?

### DISCUSSION THREAD

**Discussion by gqqnbig** [seaweedfs/seaweedfs#3110] [General]
I used to only have one volume server and I already uploaded some data.

Later I added another volume server 172.25.17.11:8080, as the figure shows.

![firefox_xhYqapeH6H](https://user-images.githubusercontent.com/614159/170854472-8cfdd9b0-01c7-4f20-9a68-d6aaf8a4ba78.png)

Next, I write new data to my weed mount point. I find 172.25.17.11:8080 still has 0 volumes. 

It's OK that [seaweedfs doesn't automatically rebalance existing data](https://github.com/chrislusf/seaweedfs#additional-features), but how about new data? Are new data balanced across multiple volume servers?

I don't like Seaweedfs continue bloating my first server and leave the second server idle.

How do I solve this problem?

**Comment by chrislusf** [comment]
run `volume.rebalance` in `weed shell`

**Reply by gqqnbig** [reply]
After that, do files evenly distribute or based on the size of volume servers?

Say volume server 1 has 10 TB, volume server 2 has 1 TB. Are files more likely be put on volume server 1?

I can't find any documentations.

**Comment by gqqnbig** [comment]
I ran this

```
> lock
> volume.balance
hdd 0.19 0.24:0.08        moving  volume 17 172.25.17.12:8080 => 172.25.17.11:8080
hdd 0.19 0.22:0.15        moving  volume 19 172.25.17.12:8080 => 172.25.17.11:8080
> unlock
>

```

But my new server still has 0 volumes.


![firefox_HNPFzlhuMh](https://user-images.githubusercontent.com/614159/170862201-98fdcdc7-75c3-4bed-bc1b-dc8402767e61.png)
 
Volume 17 and 19 are still on .12, not .11.
![firefox_fNbQgeySqV](https://user-images.githubusercontent.com/614159/170862200-cdc6eae8-104a-456c-a1d8-946fe0a095e2.png)

**Reply by chrislusf** [reply]
run `volume.rebalance -force` in `weed shell`
